File: backend/Models/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("payments")]
    public class Payment
    {
        [Key] public int Id { get; set; }

        public int OrderId { get; set; }

        [ForeignKey("OrderId")]
        public SalesOrder Order { get; set; }

        [Required]
        public decimal Amount { get; set; }

        [StringLength(20)]
        public string PaymentMethod { get; set; } = "cash";

        public DateTime PaidAt { get; set; } = DateTime.Now;
    }
}

File: backend/src/main/java/pika/hungt1/dx/entity/Payment.java
package pika.hungt1.dx.entity;

import lombok.*;
import jakarta.persistence.*;

import java.math.BigDecimal;
import java.time.Instant;
import com.fasterxml.jackson.annotation.JsonBackReference;
@Entity
@Table(name = "payments")
@Data @NoArgsConstructor @AllArgsConstructor
public class Payment {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name="order_id")
    private SalesOrder order;

    @Column(nullable=false)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    private Method paymentMethod = Method.cash;

    @Column(name="paid_at")
    private Instant paidAt = Instant.now();

    public enum Method { cash, bank_transfer, ewallet }
}
